### PR TITLE
aws-lambda-cpp: add recipe

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **aws-lambda-cpp/***

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by SSE4 — APPROVED** [review]

File: recipes/aws-lambda-cpp/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires("libcurl/8.2.1")
```
Comment (czoido) [inline recipes/aws-lambda-cpp/all/conanfile.py:48]: May we use a version range for libcurl?
```suggestion
        self.requires("libcurl/[>=7.78.0 <9]")
```

**Review by czoido — DISMISSED** [review]

File: recipes/aws-lambda-cpp/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        "shared": False,
+        "fPIC": True,
+    }
+
+    @property
+    def _min_cppstd(self):
+        return 11
+
+    def export_sources(self):
+        export_conandata_patches(self)
```
Comment (franramirez688) [inline recipes/aws-lambda-cpp/all/conanfile.py:34]: We could drop the `patch` lines as we're not applying anything (and the imports above).

File: recipes/aws-lambda-cpp/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.generate()
+        dpes = CMakeDeps(self)
+        dpes.generate()
+
+    def build(self):
+        apply_conandata_patches(self)
```
Comment (franramirez688) [inline recipes/aws-lambda-cpp/all/conanfile.py:68]: same here

File: recipes/aws-lambda-cpp/all/conanfile.py (hunk 2 above)
Comment (czoido) [inline recipes/aws-lambda-cpp/all/conanfile.py:34]:
```suggestion
```

File: recipes/aws-lambda-cpp/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
```
Comment (czoido) [inline recipes/aws-lambda-cpp/all/conanfile.py:66]:
```suggestion
```

File: recipes/aws-lambda-cpp/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,82 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.files import apply_conandata_patches, export_conandata_patches, get, copy, rmdir
```
Comment (czoido) [inline recipes/aws-lambda-cpp/all/conanfile.py:3]:
```suggestion
from conan.tools.files import get, copy, rmdir
```

**Review by czoido — APPROVED** [review]

**Review by franramirez688 — APPROVED** [review]

**Comment by chrisc66** [comment]
I am creating a project that uses aws-lambda-cpp and installing package using conan. I am not able to find the function `aws_lambda_package_target` when I install with conan. Would you please let me know if there is a workaround or solution for this problem? 
@toge

**Comment by uilianries** [comment]
@chrisc66 Please, open an issue to describe your problem with more details, and please, don't forget to fill all fields in the issue, it really helps to reproduce your scenario.

https://github.com/conan-io/conan-center-index/issues

**Pull request merged**